edge_prediction_experiment/01_hetionet_recall_dag.py

```diff
@@ -0,0 +1,348 @@
+# ---
+# jupyter:
+#   jupytext:
+#     formats: ipynb,py:light
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.9.1+dev
+#   kernelspec:
+#     display_name: Python [conda env:snorkeling_full_text]
+#     language: python
+#     name: conda-env-snorkeling_full_text-py
+# ---
+
+# # Measure Disease associations Gene Edge Recall
+
+# This notebook is designed to estimate how many hetionet edges can be recalled from this weakly supervised approach.
+# After the discriminator model has been trained, the last step is to predict whether every candidate sentence is mentions a disease associates gene relationship.
+# Following the predictions, the next step is to group each candidate sentence based on each candidate pair and take the max value of each group.
+# This max value represents the probability that an edge should or does exist.
+# Lastly, once each candidate has been scored I use the precision recall curve to estimate how many edges can be recovered from this method.
+
+# +
+# %load_ext autoreload
+# %autoreload 2
+# %matplotlib inline
+
+import math
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import plotnine as p9
+import plydata as ply
+import plydata.tidy as ply_tdy
+from sklearn.metrics import precision_recall_curve, roc_curve, auc
+from sqlalchemy import create_engine
+
+# +
+# Set up the environment
+username = "danich1"
+password = "snorkel"
+dbname = "pubmed_central_db"
+
+# Path subject to change for different os
+database_str = (
+    f"postgresql+psycopg2://{username}:{password}@/{dbname}?host=/var/run/postgresql"
+)
+conn = create_engine(database_str)
+# -
+
+hetionet_dag_map_df = pd.read_csv(
+    Path("../snorkeling_helper/label_functions/knowledge_bases")
+    / "disease_associates_gene.tsv.xz",
+    sep="\t",
+)
+hetionet_dag_map_df >> ply.slice_rows(5)
+
+sentence_prediction_df = pd.read_csv(
```

Comment:
These are all new sentences from your test set and for each sentence this is the prediction for whether or not the sentence has a gene associates gene label. For each sentence there is a prediction for multiple pairs of (gene, disease)? And you take the max prediction score for each sentence? 

This is based on my understanding of your description at the top `Following the predictions, the next step is to group each candidate sentence based on each candidate pair and take the max value of each group.`

Reply:
>These are all new sentences from your test set and for each sentence this is the prediction for whether or not the sentence has a gene associates gene label. For each sentence there is a prediction for multiple pairs of (gene, disease)? And you take the max prediction score for each sentence?

You are correct. The task of transitioning from sentences to edge type is difficult because some pairs are mentioned a lot and vise versa. This makes taking the mean or median not appropriate as the more mentioned pairs may unfairly be penalized. Hence max is the temp solution for this project.